bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/protocol/QbusCommunication.java

```diff
@@ -0,0 +1,943 @@
+                    } else if (("returnScene").equals(cmd)) {
+                        cmdListScenes(((QbusMessageListMap) qMessageGson).getOutputs());
+                    } else if (("returnCo2").equals(cmd)) {
+                        cmdListCo2(((QbusMessageListMap) qMessageGson).getOutputs());
+                    } else if (("returnRol02p").equals(cmd)) {
+                        cmdListRol(((QbusMessageListMap) qMessageGson).getOutputs());
+                    } else if (("returnSlat").equals(cmd)) {
+                        cmdListRolSlats(((QbusMessageListMap) qMessageGson).getOutputs());
+                    }
+
+                    // Incoming commands from Qbus Client to openHAB (event)
+                    else if ("updateBistabiel".equals(cmd)) {
+                        updateBistabiel(((QbusMessageListMap) qMessageGson).getOutputs());
+                    } else if ("updateDimmer".equals(cmd)) {
+                        updateDimmer(((QbusMessageListMap) qMessageGson).getOutputs());
+                    } else if ("updateThermostat".equals(cmd)) {
+                        updateThermostat(((QbusMessageListMap) qMessageGson).getOutputs());
+                    } else if ("updateCo2".equals(cmd)) {
+                        updateCO2(((QbusMessageListMap) qMessageGson).getOutputs());
+                    } else if ("updateRol02p".equals(cmd)) {
+                        updateRol(((QbusMessageListMap) qMessageGson).getOutputs());
+                    } else if ("updateRol02pSlat".equals(cmd)) {
+                        updateRolSlats(((QbusMessageListMap) qMessageGson).getOutputs());
+                    }
+
+                    // Incomming commands from Qbus server to verify the client connection
+                    else if ("disconnect".equals(cmd)) {
+                        eventDisconnect();
+                    } else if ("notConnected".equals(cmd)) {
+                        noConnection();
+                    } else if ("connected".equals(cmd)) {
+                        connection();
+                    }
+                }
+
+            } catch (JsonParseException e) {
+                logger.warn("Not acted on unsupported json {}", qMessage);
+            }
+        }
+    }
+
+    /**
+     * After setting up the communication with the Qbus Server, send all initialization messages.
+     * <p>
+     * First send connect to connect with the Qbus Server application
+     * Get request for Bistabiel/Timers/Intervals/Mono outputs
+     * Get request for the Scenes
+     * Get request for Dimmers 1T and 2T
+     * Get request for Shutters
+     * Get request for Thermostats
+     * Get request for CO2
+     */
+    @Override
+    public void initialize() {
+        QbusBridgeHandler handler = bridgeCallBack;
+
+        if (bridgeCallBack != null) {
+            if (ctdConnected) {
+                try {
+                    sendAndReadMessage("getBistabiel");
```

Comment:
`initialize()` must return fast. As this will call some network code, you need to execute it asynchronously e.g. via `scheduler.submit()`.